gprofiler/profilers/python.py

```diff
@@ -36,12 +39,60 @@
 
 logger = get_logger_adapter(__name__)
 
+_module_name_in_stack = re.compile(r"\((?P<module_info>(?P<filename>[^\)]+?\.py):\d+)\)")
+
+
+def _add_versions_to_process_stacks(process: Process, stacks: StackToSampleCount) -> StackToSampleCount:
+    new_stacks: StackToSampleCount = Counter()
+    for stack in stacks:
+        modules_paths = (match.group("filename") for match in _module_name_in_stack.finditer(stack))
+        packages_versions = get_modules_versions(modules_paths, process)
+
+        def _replace_module_name(module_name_match):
+            package_info = packages_versions.get(module_name_match.group("filename"))
+            if package_info is not None:
+                package_name, package_version = package_info
+                return "({} ({}-{}))".format(module_name_match.group("module_info"), package_name, package_version)
+            return module_name_match.group()
+
+        new_stack = _module_name_in_stack.sub(_replace_module_name, stack)
+        new_stacks[new_stack] = stacks[stack]
+
+    return new_stacks
+
+
+def _add_versions_to_stacks(
+    process_to_stack_sample_counters: ProcessToStackSampleCounters,
+) -> ProcessToStackSampleCounters:
+    result: ProcessToStackSampleCounters = defaultdict(Counter)
+
+    for pid, stack_to_sample_count in process_to_stack_sample_counters.items():
+        try:
+            process = Process(pid)
+        except NoSuchProcess:
+            # The process doesn't exist anymore so we can't analyze versions
+            continue
+        result[pid] = _add_versions_to_process_stacks(process, stack_to_sample_count)
+
+    return result
+
 
 class PySpyProfiler(ProcessProfilerBase):
     MAX_FREQUENCY = 50
     _BLACKLISTED_PYTHON_PROCS = ["unattended-upgrades", "networkd-dispatcher", "supervisord", "tuned"]
     _EXTRA_TIMEOUT = 10  # give py-spy some seconds to run (added to the duration)
 
+    def __init__(
+        self,
+        frequency: int,
+        duration: int,
+        stop_event: Optional[Event],
+        storage_dir: str,
+        add_versions: bool,
```

Comment:
Please make this a kwarg-only. It is unclear when it's only a `bool` specified without the name.

Reply:
The `add_versions`